highcharts 10.3.2 + es-modules breaks Angular projects (`Cannot find name 'AnyRecord'`)

#### Expected behaviour
Successful compilation

#### Actual behaviour
Compilation fails

```bash
Error: node_modules/highcharts/es-modules/masters/highcharts.src.d.ts:9:18 - error TS2304: Cannot find name 'AnyRecord'.

9 declare const G: AnyRecord;
                   ~~~~~~~~~




** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **


✖ Failed to compile.
```
<img width="858" alt="Screenshot 2022-11-28 at 14 47 34" src="https://user-images.githubusercontent.com/1189149/204330152-76024b9d-de4f-4504-a313-d95d735b8841.png">


#### Live demo with steps to reproduce
Could not reproduce it with a StackBlitz demo so I'm attaching an Angular project instead: [demo-app.zip](https://github.com/highcharts/highcharts/files/10105651/demo-app.zip)

To run the demo app open it's folder on a terminal/command line, run `npm install` and run `npx ng serve -o`.


#### Product version
highcharts 10.3.2

#### Affected browser(s)
All

--- a/.husky/pre-commit
+++ b/.husky/pre-commit
@@ -1,4 +1,9 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
 
-npx lint-staged && npx gulp lint-ts && npx gulp scripts && npx ts-node ./test/ts-node-unit-tests && npx gulp test && npx gulp test-ts
+npx lint-staged && \
+npx gulp lint-ts && \
+npx gulp scripts && \
+npx ts-node ./test/ts-node-unit-tests && \
+npx gulp test && \
+npx gulp test-ts

--- a/package.json
+++ b/package.json
@@ -20,7 +20,8 @@
   ],
   "scripts": {
     "test": "npx gulp test-ts --force && npx gulp test --force --speak",
-    "testall": "npx gulp test --browsers all --force --speak",
+    "test-all": "npx gulp test --browsers all --force --speak",
+    "test-dts": "npx gulp scripts && npx gulp jsdoc-dts && npx gulp lint-dts",
     "build": "npx gulp dist",
     "jsdoc": "npx gulp jsdoc-watch --skip-websearch",
     "jsdoc-highcharts": "npx gulp jsdoc-watch --products highcharts --skip-websearch",

--- a/test/typescript-dts/es-modules/masters.ts
+++ b/test/typescript-dts/es-modules/masters.ts
@@ -1,0 +1,9 @@
+import Highcharts from 'highcharts/es-modules/masters/highcharts.src.js';
+import HighchartsGantt from 'highcharts/es-modules/masters/highcharts-gantt.src.js';
+import HighchartsMaps from 'highcharts/es-modules/masters/highmaps.src.js';
+import HighchartsStock from 'highcharts/es-modules/masters/highstock.src.js';
+
+Highcharts.chart({});
+HighchartsGantt.chart({});
+HighchartsMaps.mapChart({});
+HighchartsStock.stockChart({});

--- a/test/typescript-dts/es-modules/tsconfig.json
+++ b/test/typescript-dts/es-modules/tsconfig.json
@@ -1,0 +1,7 @@
+{
+    "extends": "../tsconfig.json",
+    "compilerOptions": {
+        "module": "ESNext",
+        "moduleResolution": "NodeNext"
+    }
+}

--- a/tools/gulptasks/scripts-js.js
+++ b/tools/gulptasks/scripts-js.js
@@ -49,7 +49,13 @@ function task() {
         processLib.isRunning('scripts-js', true);
 
         BuildScripts
+            // assemble JS files
             .fnFirstBuild()
+            // deleting invalid masters DTS
+            .then(() => fsLib.getFilePaths('js/masters/', true).forEach(
+                path => path.endsWith('.d.ts') && fsLib.deleteFile(path)
+            ))
+            // copy valid native DTS
             .then(() => fsLib.copyAllFiles(
                 'js/',
                 'code/es-modules/',

